### PR TITLE
[SPARK-41426][UI] Protobuf serializer for ResourceProfileWrapper

### PR DESCRIPTION
**Pull request by techaddict**
### What changes were proposed in this pull request?
Add Protobuf serializer for ResourceProfileWrapper

### Why are the changes needed?
Support fast and compact serialization/deserialization for ResourceProfileWrapper over RocksDB.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
New UT

### CODE REVIEW

**Comment by techaddict** [comment]
@gengliangwang This is ready for review

File: core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala [hunk 1]
```diff
@@ -458,4 +458,46 @@ class KVStoreProtobufSerializerSuite extends SparkFunSuite {
...
+        id = 0,
+        executorResources = Map(
+          "0" -> new ExecutorResourceRequest(
+            resourceName = "exec1",
+            amount = 1,
+            discoveryScript = "script0",
+            vendor = "apache"),
+          "1" -> new ExecutorResourceRequest(
+            resourceName = "exec2",
+            amount = 1,
```
Comment (gengliangwang) [inline core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala:474]: let's use a different value

File: core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala [hunk 2]
```diff
@@ -458,4 +458,46 @@ class KVStoreProtobufSerializerSuite extends SparkFunSuite {
...
+            discoveryScript = "script1",
+            vendor = "apache")
```
Comment (gengliangwang) [inline core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala:476]: let's use a different value?

**Comment by techaddict** [comment]
@gengliangwang addressed comments

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**